tortoisebeacon/handlers.go

```diff
@@ -473,19 +469,61 @@ func (tb *TortoiseBeacon) storeFollowingVotes(message FollowingVotingMessage, mi
 		}
 	}
 
+	// TODO(kimmy): keep later rounds votes in a separate buffer so we don't count them prematurely
+	// tho i am not sure whether counting votes for later rounds early is a security concern.
 	for vote := range thisRoundVotes.invalid {
 		if _, ok := tb.votesMargin[vote]; !ok {
 			tb.votesMargin[vote] = new(big.Int).Neg(voteWeight)
 		} else {
 			tb.votesMargin[vote].Sub(tb.votesMargin[vote], voteWeight)
 		}
 	}
-
-	tb.hasVoted[message.RoundID][string(minerPK.Bytes())] = struct{}{}
 }
 
 func (tb *TortoiseBeacon) currentEpoch() types.EpochID {
 	tb.mu.RLock()
 	defer tb.mu.RUnlock()
 	return tb.epochInProgress
 }
+
+func (tb *TortoiseBeacon) registerProposed(minerPK *signing.PublicKey, logger log.Log) error {
```

Comment:
how do we handle local proposals now? relying exclusively on the self-broadcast to receive them?

Reply:
yes. is there a possibility that self-broadcast won't work?

Reply:
just wanted to clarify. i don't like the pattern much as it seems wasteful to me, but nothing besides that